### ISSUE TITLE
Dealing with adds

### ISSUE THREAD

**Issue by Dragoo-beep** [Xian55/WowClassicGrindBot#651]
### Description

Wondering what im doing wrong, if my pet pulls 2 mobs we just stand and wait for pet to die or mob to die before we continue.

[13:46:36:496 I] [GoapAgent        ] Kill credit detected! Session Total: 7 | Last Combat: 1 | Currently fighting: 1
[13:46:36:529 I] [CombatGoal       ] Lost target!
[13:46:36:544 W] [GoapAgent        ] New Plan= NO PLAN

I have 
Assist Target -> F by default is game settings
Target Pet -> Numpad Multiply has to setup manually
Pet Attack -> Numpad Subtract has to setup manually
This works as if I stop targeting the current target im fighting he retargets via pet instantly.

Gone over the settings many times but im not getting any feedback of what we are trying to do, so im not sure whats happening.

### Addon Version

1.7.60

### World of Warcraft Client

World of Warcraft Classic

### Reproduction Steps

Pull a mob, then pull another, kill first mob, then we know the second target exists but we have no plan.

### Logs

[headless_out20241231_003.log](https://github.com/user-attachments/files/18281664/headless_out20241231_003.log)

**Comment by Xian55** [comment]
Gooday!

> Wondering what im doing wrong, if my pet pulls 2 mobs we just stand and wait for pet to die or mob to die before we continue.

After i reading the following i have a hunch that you might be set your pet behaviour to `Pet Passive` by any chance.
Setting your pet to [Defensive](https://wowwiki-archive.fandom.com/wiki/Defensive) should resolve the issue.

I'm pretty sure that once you pull more then one target with your pet - accidental body pull even - once the primary target dies, your pet should target the next potential enemy as long as your pet behaviour is set to `Defensive`.

There's a GoapGoal for it called [TargetPetTargetGoal](https://github.com/Xian55/WowClassicGrindBot/blob/db0d8fadc681b34e275a2312c0717938e047411c/Core/Goals/TargetPetTargetGoal.cs#L5), which is purposefully designed to take advantage of the `Defensive` behaviour.

As long as your character class properly extracted, and read by the backend, this [Goal should be added](https://github.com/Xian55/WowClassicGrindBot/blob/db0d8fadc681b34e275a2312c0717938e047411c/Core/GoalsFactory/GoalFactory.cs#L249).

Let me know if the issue still persists.

---

On the other hand, the following is unrelated to the original issue, however two things caught my eye and they might hinder you to have the expected experience with the project.

> [13:43:51:981 I] [WowScreenDXGI    ] Rectangle [ X=961, Y=547, Width=958, Height=484 ] - Windowed Mode: True - Scale: 1,00 - Monitor Rect: RawRect { Left = 1920, Top = -5, Right = 3840, Bottom = 1075 } - Monitor Index: 1

* That **Top = -5** hints that, you might have multiple monitors and there might be an overlap between the top and bottom monitors?
* `Width=958, Height=484` i would recommend to stick with the pre-set resolutions, the closest would be like `800x600`, as the in game ui uses scaling, in the past i've experienced wierd behaviours due overlapping/overdrawn pixels, which resulted non deterministic strange behaviours.

Such as
```cs
[14:00:58:754 I] [GoapAgent        ] New Plan=  Parallel
[14:00:58:773 E] Index was outside the bounds of the array.
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Core.ActionBarCostReader.Update(IAddonDataProvider reader) in C:\Users\S\Documents\L\L\Core\Actionbar\ActionBarCostReader.cs:line 48
   at Core.AddonReader.Update() in C:\Users\S\Documents\L\L\Core\Addon\AddonReader.cs:line 73
   at Core.BotController.AddonThread() in C:\Users\S\Documents\L\L\Core\BotController.cs:line 156
```

In the logs, i saw many negative numbers regarding the actions await times. Like
```
 184: [13:44:02:588 I] [CastingHandler   ] [Serpent Sting    ] castbar input 61ms -528,5215ms
 399: [13:46:47:474 I] [CastingHandler   ] [Mend Pet         ] castbar input 77ms -487,1486ms
```

Where you can see two elapsed milliseconds notice
* the first one is how long does the bound key was pressed, for `Mend Pet` it was `77` milliseconds
* the second is how long does it took for the c# backend to detect the keypress effect, if that number negative, then it means that the c# backend was unable read the changes by the addon. 

You can think of it as like a cause and effect.
* You press a button
* after the button press something going to happen and that happening was not properly propagated from the addon to the c# backend.

---

I could give you a few hints what you can look up like
* enforcing fixed framerate/fps in-game, to have consistent game refresh rate you can pick (15-30-60-90-120)
* disable v-sync - it introduces a big keypress to display update which is hinders the wow addon -> c# backend communication.

**Comment by Dragoo-beep** [comment]
Tyvm for the great answer! Pet is on defensive and he is fighting next mob while we are waiting. I did some changes to code where if our pet has a target and we dont, that we always assist pet, and that seems to work. I fixed the -5 offset aswell but doesnt seem to help. I also tried all the resolutions with different cell sizes.

I also looked into the negative numbers and I tested it on 3 different computers they all have negative numbers, and my friend has the same on his computer (I didnt help him to set it up so he must have done exact same mistake). 

I tested with capping fps to 30, 60, 120 etc, I tried changing hertz and v-sync is also disabled.

Im guessing the same bug makes us engage in pvp randomly aswell despite a strict "AllowPvP": false.

[10:45:11:659 W] [TargetBlacklist  ] (0,17035,MINO) is player or pet!
[10:45:11:660 I] [GoapAgent        ] New Plan=  Blacklist Target
[10:45:11:752 I] [GoapAgent        ] New Plan=  Follow  Arathi  Highlands_2025_01_05_13_02_17
[10:45:11:753 I] [Navigation       ] [LocalPathingApi] distance is close 7,690509. Keep RouteToWaypoint.
[10:45:12:098 I] [GoapAgent        ] New Plan=  Pull Target
[10:45:12:145 I] [PullTargetGoal   ] Stop auto interact!
[10:45:12:178 I] [FollowRouteGoal  ] Found target!
[10:45:12:503 W] [CastingHandler   ] [Hunter's Mark    ] ... Failed try again!
[10:45:12:504 I] [ReactCastError   ] React to SPELL_FAILED_NOT_READY -- wait until its ready
[10:45:12:753 I] [PullTargetGoal   ] Preventing pulling possible tagged target!

We engage about 30% of enemy players.

**Comment by Xian55** [comment]
Gooday.

> Tyvm for the great answer! Pet is on defensive and he is fighting next mob while we are waiting. I did some changes to code where if our pet has a target and we dont, that we always assist pet, and that seems to work.

It would be great if you could share the mentioned changes in a PR or just sending a diff about the specific changes, it would help a project a lot.

> I also looked into the negative numbers and I tested it on 3 different computers they all have negative numbers, and my friend has the same on his computer (I didnt help him to set it up so he must have done exact same mistake).

In the past, i've seen user logs who had similar negative numbers, however i was yet to figure out what would case it. Personally i have never seen such issue, i mean the volume of it, it may happens once of twice but not that consistency what your log presented.

> Im guessing the same bug makes us engage in pvp randomly aswell despite a strict "AllowPvP": false.

So you are trying to tell me that, even tho the `"AllowPvP": "false"` is set, the bot still actively engaging enemy players. Keep in mind that once an enemy player is engaging you, the bot will fight back.

**Comment by Dragoo-beep** [comment]
[CombatGoal.zip](https://github.com/user-attachments/files/18322553/CombatGoal.zip)
I just changed some stuff in combatgoal with chatbot so this will need some work as its pretty scuffed, but it beat just watching pet die.

> So you are trying to tell me that, even tho the "AllowPvP": "false" is set, the bot still actively engaging enemy players. Keep in mind that once an enemy player is engaging you, the bot will fight back.

Yes exactly :P We actively engage in pvp. This happens to my friend aswell he added players to the blacklist for a mega scuffed fix.

**Comment by Xian55** [comment]
> [CombatGoal.zip](https://github.com/user-attachments/files/18322553/CombatGoal.zip) I just changed some stuff in combatgoal with chatbot so this will need some work as its pretty scuffed, but it beat just watching pet die.
> 
> > So you are trying to tell me that, even tho the "AllowPvP": "false" is set, the bot still actively engaging enemy players. Keep in mind that once an enemy player is engaging you, the bot will fight back.
> 
> Yes exactly :P We actively engage in pvp. This happens to my friend aswell he added players to the blacklist for a mega scuffed fix.

If you were kind enough to provide logs where the mentioned issue exists, i might be able to see what went wrong, thanks!

**Comment by Dragoo-beep** [comment]
The above log is from when we engaged a player;
> [10:45:11:659 W] [TargetBlacklist ] (0,17035,MINO) is player or pet!
[10:45:11:660 I] [GoapAgent ] New Plan= Blacklist Target
[10:45:11:752 I] [GoapAgent ] New Plan= Follow Arathi Highlands_2025_01_05_13_02_17
[10:45:11:753 I] [Navigation ] [LocalPathingApi] distance is close 7,690509. Keep RouteToWaypoint.
[10:45:12:098 I] [GoapAgent ] New Plan= Pull Target
[10:45:12:145 I] [PullTargetGoal ] Stop auto interact!
[10:45:12:178 I] [FollowRouteGoal ] Found target!
[10:45:12:503 W] [CastingHandler ] [Hunter's Mark ] ... Failed try again!
[10:45:12:504 I] [ReactCastError ] React to SPELL_FAILED_NOT_READY -- wait until its ready
[10:45:12:753 I] [PullTargetGoal ] Preventing pulling possible tagged target!

I will save a full log for next time I see him engage in pvp.

**Comment by Dragoo-beep** [comment]
[headless_out20250109.log](https://github.com/user-attachments/files/18358586/headless_out20250109.log)
Caught him chasing a warrior through badlands lol. Other things to note is the frontend not showing our coordinate in picture and the preview is black. I tried reinstalling game but I think its time to reformat computer instead. This happens only on main computer that is probably messed up somehow.
![image](https://github.com/user-attachments/assets/56135f9e-0eb4-479c-b1fb-bfeb42ef61b2)
and some addon errors;
![image](https://github.com/user-attachments/assets/a45fb21b-e60e-4954-8ead-9d4560ca9b02)
4x [ADDON_ACTION_BLOCKED] AddOn 'tsm' tried to call the protected function 'UNKNOWN()'.
[string "@!BugGrabber/BugGrabber.lua"]:485: in function <!BugGrabber/BugGrabber.lua:485>
[string "=[C]"]: ?
[string "=[C]"]: ?
[string "@tsm/Libs/LibRangeCheck-2.0-100204/LibRangeCheck-2.0.lua"]:458: in function `checker'
[string "@tsm/Libs/LibRangeCheck-2.0-100204/LibRangeCheck-2.0.lua"]:583: in function <...Ons/tsm/Libs/LibRangeCheck-2.0/LibRangeCheck-2.0.lua:578>
[string "=(tail call)"]: ?
[string "@tsm/Query.lua"]:324: in function `getRange'
[string "@tsm/tsm.lua"]:686: in function <tsm/tsm.lua:508>

Locals:
_ = Frame {
}
event = "ADDON_ACTION_BLOCKED"
events = <table> {
}
12x BindPad/BindPad.xml:657 Unrecognized XML attribute: bottom
12x BindPad/BindPad.xml:657 Unrecognized XML attribute: top
12x BindPad/BindPad.xml:657 Unrecognized XML attribute: right
12x BindPad/BindPad.xml:657 Unrecognized XML attribute: left
12x BindPad/BindPad.xml:657 Unrecognized XML: AbsInset
12x BindPad/BindPad.xml:656 Unrecognized XML: BackgroundInsets
12x BindPad/BindPad.xml:655 Unrecognized XML attribute: tile
12x BindPad/BindPad.xml:655 Unrecognized XML attribute: edgeFile
12x BindPad/BindPad.xml:655 Unrecognized XML attribute: bgFile
12x BindPad/BindPad.xml:655 Unrecognized XML: Backdrop

**Comment by Xian55** [comment]
> [headless_out20250109.log](https://github.com/user-attachments/files/18358586/headless_out20250109.log)

It would be nice if you could attach the first part of the logs, it seems when you were running the project you progressed to "tomorrow" so the yesterday log have not been included. When you start the project there are key information logged about the environment. They are crucial for me to understand whats going on!

---

>  Other things to note is the frontend not showing our coordinate in picture and the preview is black

That is by design, that element in the frontend only shows images when the screen is getting captured for the NpcNameFinder component. So its not designed to show a real time status of the game. It was a performance choice of mine. I have not yet found a way to stream bitmaps efficiently. 

---

Regarding to the Bindpad, just ignore them, they are only warning.

---

> 4x [ADDON_ACTION_BLOCKED] AddOn 'tsm' tried to call the protected function 'UNKNOWN()'.
[string "@!BugGrabber/BugGrabber.lua"]:485: in function <!BugGrabber/BugGrabber.lua:485>
[string "=[C]"]: ?
[string "=[C]"]: ?
[string "@tsm/Libs/LibRangeCheck-2.0-100204/LibRangeCheck-2.0.lua"]:458: in function checker' [string "@tsm/Libs/LibRangeCheck-2.0-100204/LibRangeCheck-2.0.lua"]:583: in function <...Ons/tsm/Libs/LibRangeCheck-2.0/LibRangeCheck-2.0.lua:578> [string "=(tail call)"]: ? [string "@tsm/Query.lua"]:324: in function getRange'
[string "@tsm/tsm.lua"]:686: in function <tsm/tsm.lua:508>

By any chance are you using the custom addon name as `tsm` ?

It seems blizzard has pushed some changes which prevents `LibRangeCheck-2.0` to work properly.

On the other hand, `LibRangeCheck-2.0` is no longer receives updates since last year, might have to look into integrating `LibRangeCheck-3.0`

> [IsItemInRange](https://wowpedia.fandom.com/wiki/API_IsItemInRange)() is a restricted function as of 2023 Nov 16.

Hah it seems like this issue exists for a long time 
https://www.reddit.com/r/classicwow/comments/17xgw8n/range_checker_was_hotfixed_out_of_the_addon_api/

**Comment by Xian55** [comment]
> 4x [ADDON_ACTION_BLOCKED] AddOn 'tsm' tried to call the protected function 'UNKNOWN()'.

Should be fixed Since #653

**Comment by Xian55** [comment]
After #654 i would hope that the blacklist target action going to be respected among the stack. 

Let me know how it turns out!

**Comment by Dragoo-beep** [comment]
> By any chance are you using the custom addon name as tsm ? 
Yeah im so clever :PP

Full log: 
[headless_out20250106.log](https://github.com/user-attachments/files/18361985/headless_out20250106.log)
[headless_out20250107.log](https://github.com/user-attachments/files/18361988/headless_out20250107.log)
[headless_out20250108.log](https://github.com/user-attachments/files/18361989/headless_out20250108.log)
[headless_out20250109.log](https://github.com/user-attachments/files/18361990/headless_out20250109.log)

Testing the new update now! 

![image](https://github.com/user-attachments/assets/2d004bee-e69f-4bfe-90b4-b2a3a52e1502)
Here btw I ment that the cursor is stuck in top left corner where it should be moving around the waypoints as it does on my laptop.

**Comment by Xian55** [comment]
Thank you for the longs!

>Here btw I ment that the cursor is stuck in top left corner where it should be moving around the waypoints as it does on my laptop.

Okay i see what you mean now. 

Can you send an image which contains the top right side of that `Route` component?

There we can see the coordinates where the player is currently at.
like this
![image](https://github.com/user-attachments/assets/d56bb306-d08d-460f-a963-7900e56cd58f)

**Comment by Dragoo-beep** [comment]
![image](https://github.com/user-attachments/assets/87a601d0-8ee3-4c1e-aa44-b30d459e7ced)
Navigation seems to work just fine and I only use headless anyway just thought it was worth pointing out.

it also knows what point it is on
![image](https://github.com/user-attachments/assets/37b33662-a980-48d7-88bd-7beb60217070)

**Comment by Xian55** [comment]
I suspect there might be some Culture related issue. 

If get it correctly, at your culture the floating point numbers are represented with `,` character separator. like `23,235`

For mine, we are using `.` as a separator. Like `23.235`

**Comment by Dragoo-beep** [comment]
> I suspect there might be some Culture related issue.
> 
> If get it correctly, at your culture the floating point numbers are represented with `,` character separator. like `23,235`
> 
> For mine, we are using `.` as a separator. Like `23.235`

Maybe its why '=' does not work for me as keybind aswell

**Comment by Xian55** [comment]
If you want to use the `=` keybind, you have set your input language to `ENG`.

Mentioned in the readme [10.1. Actionbar Key Bindings:](https://github.com/Xian55/WowClassicGrindBot?tab=readme-ov-file#101-actionbar-key-bindings:~:text=setup%20while%20using-,English%20Keyboard,-layout.)

**Comment by Dragoo-beep** [comment]
> I suspect there might be some Culture related issue.
> 
> If get it correctly, at your culture the floating point numbers are represented with `,` character separator. like `23,235`
> 
> For mine, we are using `.` as a separator. Like `23.235`

![image](https://github.com/user-attachments/assets/5ccc51b8-592e-4edf-aa82-f73e3bd35a58)

Genius 👍  I changed from norwegian to english on system and voila!

**Comment by Dragoo-beep** [comment]
Though still not dealing with adds. Pet is fighting mob while we wait for him to kill it.
![image](https://github.com/user-attachments/assets/d409034b-651e-4344-a6d4-510a4cc078db)
Unless I do my wonky fix
[out20250109_001.log](https://github.com/user-attachments/files/18362465/out20250109_001.log)

**Comment by Xian55** [comment]
About that, i haven't had the time to look into it as of yet. Recently some changes where happened in most likely it got broke.

**Comment by Dragoo-beep** [comment]
Still seem to attack players at random, but later we ignored the same player so its an improvement
[headless_out20250109.log](https://github.com/user-attachments/files/18363979/headless_out20250109.log)

**Comment by Xian55** [comment]
Alright thanks for the feedback, the search for new target related code is in a different thread, so i have to find a way to keep them in sync.

**Comment by Xian55** [comment]
I've did some testing and at my end it works as expected.

At the readme i would like to point out two important setup requirement
* [9. Configure the Wow Client - Key Bindings](https://github.com/Xian55/WowClassicGrindBot?tab=readme-ov-file#9-configure-the-wow-client---key-bindings) - **Target Pet** and **Assist Target** and **Pet attack**
* [11. Configure the Wow Client - Bindpad addon](https://github.com/Xian55/WowClassicGrindBot?tab=readme-ov-file#11-configure-the-wow-client---bindpad-addon)

I'm testing at the Classis Era PTR, with 60lvl hunter, pet is set to `defensive`
Pulling one mob with my pet then once it pulled i send my pet to the next to pull the second.
Once the first mob is dead, and as soon as my pet is getting hit by the mob my player reacts to it by
* targeting pet
* assist target
* this makes my target to the enemy

```
[21:53:20:733 I] [GoapAgent        ] New Plan=  Pull Target
[21:53:20:752 I] [PullTargetGoal   ] Stop auto interact!
[21:53:20:806 I] [WowProcessInput  ] [Delete] press random 53ms
[21:53:20:806 V] [ConfigurableInput] [StopAttack] Delete pressed 53ms
[21:53:20:940 I] [WowProcessInput  ] [Delete] press random 57ms
[21:53:20:941 V] [ConfigurableInput] [StopAttack] Delete pressed 57ms
[21:53:20:969 I] [WowProcessInput  ] [Subtract] press random 26ms
[21:53:20:969 V] [ConfigurableInput] [PetAttack] Subtract pressed 26ms
[21:53:20:970 V] [WowProcessInput  ] [UpArrow] move Pressed 2ms
[21:53:21:000 V] [WowProcessInput  ] [UpArrow] move Pressed 3ms
[21:53:21:016 V] [WowProcessInput  ] [UpArrow] move Pressed 3ms
[21:53:21:064 V] [WowProcessInput  ] [UpArrow] move Pressed 3ms
[21:53:21:100 V] [WowProcessInput  ] [UpArrow] move Pressed 3ms
[21:53:21:150 V] [WowProcessInput  ] [UpArrow] move Pressed 3ms
[21:53:21:198 V] [WowProcessInput  ] [UpArrow] move Pressed 3ms
[21:53:21:250 V] [WowProcessInput  ] [UpArrow] move Pressed 4ms
[21:53:21:317 I] [WowProcessInput  ] [I] press random 19ms
[21:53:21:317 V] [ConfigurableInput] [Approach] I pressed 19ms
[21:53:21:668 V] [WowProcessInput  ] [UpArrow] move Pressed 2ms
[21:53:21:781 I] [WowProcessInput  ] [I] press random 11ms
[21:53:21:781 V] [ConfigurableInput] [Approach] I pressed 11ms
[21:53:22:229 I] [WowProcessInput  ] [I] press random 26ms
[21:53:22:229 V] [ConfigurableInput] [Approach] I pressed 26ms
[21:53:22:686 I] [WowProcessInput  ] [I] press random 34ms
[21:53:22:687 V] [ConfigurableInput] [Approach] I pressed 34ms
[21:53:23:084 D] [CombatUtil       ] Combat Enter
[21:53:23:087 I] [GoapAgent        ] New Plan=  Combat
[21:53:23:117 V] [WowProcessInput  ] [UpArrow] move Pressed 3ms
[21:53:23:225 I] [WowProcessInput  ] [NumPad2] press random 55ms
[21:53:23:225 D] [ConfigurableInput] [Bestial Wrath] NumPad2 pressed 55ms
[21:53:23:485 I] [CastingHandler   ] [Bestial Wrath    ] castbar input 56ms -259.7757ms
[21:53:23:549 I] [CombatGoal       ] Turning too fast!
[21:53:23:549 V] [WowProcessInput  ] [UpArrow] move Pressed 4ms
[21:53:24:405 I] [WowProcessInput  ] [D3] press random 53ms
[21:53:24:406 D] [ConfigurableInput] [Auto Shot] D3 pressed 53ms
[21:53:24:667 W] [CastingHandler   ] [Auto Shot        ] ... Failed try again!
[21:53:24:667 V] [WowProcessInput  ] [UpArrow] move Pressed 2ms
[21:53:24:761 I] [WowProcessInput  ] [D3] press random 59ms
[21:53:24:761 D] [ConfigurableInput] [Auto Shot] D3 pressed 59ms
[21:53:25:419 I] [CastingHandler   ] [Auto Shot        ] castbar input 60ms -656.9984ms
[21:53:25:497 I] [WowProcessInput  ] [Subtract] press random 25ms
[21:53:25:498 V] [ConfigurableInput] [PetAttack] Subtract pressed 25ms
[21:53:25:498 V] [WowProcessInput  ] [UpArrow] move Pressed 2ms
[21:53:25:593 I] [WowProcessInput  ] [D3] press random 56ms
[21:53:25:594 D] [ConfigurableInput] [Auto Shot] D3 pressed 56ms
[21:53:26:037 W] [CastingHandler   ] [Auto Shot        ] ... Failed try again!
[21:53:26:037 V] [WowProcessInput  ] [UpArrow] move Pressed 4ms
[21:53:26:207 I] [WowProcessInput  ] [D3] press random 71ms
[21:53:26:208 D] [ConfigurableInput] [Auto Shot] D3 pressed 71ms
[21:53:26:872 I] [CastingHandler   ] [Auto Shot        ] castbar input 74ms -663.288ms
[21:53:27:265 I] [WowProcessInput  ] [D3] press random 64ms
[21:53:27:266 D] [ConfigurableInput] [Auto Shot] D3 pressed 64ms
[21:53:27:522 I] [CastingHandler   ] [Auto Shot        ] castbar input 66ms -255.4058ms
[21:53:28:386 I] [BagChangeTracker ] 168 -> 167 Ice Threaded Arrow
[21:53:31:029 I] [WowProcessInput  ] [NumPad4] press random 57ms
[21:53:31:030 D] [ConfigurableInput] [Intimidation] NumPad4 pressed 57ms
[21:53:31:286 I] [CastingHandler   ] [Intimidation     ] castbar input 59ms -255.2802ms
[21:53:31:386 I] [BagChangeTracker ] 167 -> 166 Ice Threaded Arrow
[21:53:32:891 I] [BagChangeTracker ] 166 -> 165 Ice Threaded Arrow
[21:53:34:741 I] [BagChangeTracker ] 165 -> 164 Ice Threaded Arrow
[21:53:37:473 I] [BagChangeTracker ] 164 -> 163 Ice Threaded Arrow
[21:53:37:926 I] [GoapAgent        ] Kill credit detected! Session Total: 7 | Last Combat: 1 | Currently fighting: 1
[21:53:37:929 I] [GoapAgent        ] New Plan=  Wait
[21:53:37:930 I] [WaitGoal         ] ...
[21:53:38:377 I] [GoapAgent        ] New Plan=  Target Pet Target
[21:53:38:431 I] [WowProcessInput  ] [Multiply] press random 53ms
[21:53:38:432 V] [ConfigurableInput] [TargetPet] Multiply pressed 53ms
[21:53:38:494 I] [WowProcessInput  ] [F] press random 61ms
[21:53:38:495 V] [ConfigurableInput] [TargetTargetOfTarget] F pressed 61ms
[21:53:38:558 I] [GoapAgent        ] New Plan=  Combat
[21:53:38:758 I] [WowProcessInput  ] [D3] press random 65ms
[21:53:38:759 D] [ConfigurableInput] [Auto Shot] D3 pressed 65ms
[21:53:38:991 I] [CastingHandler   ] [Auto Shot        ] castbar input 66ms 231.6748ms
[21:53:40:792 I] [BagChangeTracker ] 163 -> 162 Ice Threaded Arrow
[21:53:43:608 I] [BagChangeTracker ] 162 -> 161 Ice Threaded Arrow
[21:53:46:462 I] [BagChangeTracker ] 161 -> 160 Ice Threaded Arrow
[21:53:49:633 I] [BagChangeTracker ] 160 -> 159 Ice Threaded Arrow
[21:53:52:499 I] [BagChangeTracker ] 159 -> 158 Ice Threaded Arrow
[21:53:52:749 I] [GoapAgent        ] Kill credit detected! Session Total: 8 | Last Combat: 2 | Currently fighting: 0
[21:53:52:785 I] [GoapAgent        ] New Plan=  Consume Corpse
[21:53:52:786 I] [ConsumeCorpseGoal] Safe to consume a corpse.
[21:53:52:789 I] [GoapAgent        ] New Plan=  Loot
[21:53:52:819 I] [LootGoal         ] Lost target 29.853ms
[21:53:52:956 I] [LootGoal         ] Look at possible closest corpse and try once again...
[21:53:53:268 I] [WowProcessInput  ] [G] press random 70ms
[21:53:53:269 V] [ConfigurableInput] [TargetLastTarget] G pressed 70ms
[21:53:53:300 I] [LootGoal         ] Keyboard last target found!
[21:53:53:337 I] [WowProcessInput  ] [I] press random 37ms
[21:53:53:416 I] [WowProcessInput  ] [I] press random 33ms
[21:53:53:875 I] [WowProcessInput  ] [I] press random 40ms
[21:53:54:330 I] [WowProcessInput  ] [I] press random 44ms
[21:53:54:793 I] [WowProcessInput  ] [I] press random 41ms
[21:53:55:257 I] [WowProcessInput  ] [I] press random 54ms
[21:53:55:519 I] [LootGoal         ] Has target ? True | Reached corpse ? 2135.8425ms
[21:53:55:716 I] [WowProcessInput  ] [I] press random 49ms
[21:53:56:070 I] [BagChangeTracker ]  2 ->  3 Dark Iron Scraps
[21:53:56:071 I] [LootGoal         ] Loot Successful 551.3087ms
```

**Comment by Xian55** [comment]
Here's another scenario with video

https://github.com/user-attachments/assets/f04c6c5f-5ce3-4f22-bece-060d8fabc10c

**Comment by Xian55** [comment]
> Though still not dealing with adds. Pet is fighting mob while we wait for him to kill it. ![image](https://private-user-images.githubusercontent.com/193331462/401565727-d409034b-651e-4344-a6d4-510a4cc078db.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY2MDc0NjAsIm5iZiI6MTczNjYwNzE2MCwicGF0aCI6Ii8xOTMzMzE0NjIvNDAxNTY1NzI3LWQ0MDkwMzRiLTY1MWUtNDM0NC1hNmQ0LTUxMGE0Y2MwNzhkYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDExMVQxNDUyNDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yMDg4YjhjYmNkODdjOTVhYzU5MmRlOTc4MWU1ZTEwZjk0ZWJlODI3ZGJhMWE3NWJlOGExNDhlOWM0ZWJiMWJjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.FOQQAF600qy7NcFR5ra845H5I78q_qNlAaTG8orMqTs) Unless I do my wonky fix [out20250109_001.log](https://github.com/user-attachments/files/18362465/out20250109_001.log)

The situation should improve after #659 and 11d6efc4ff1363ef7744d6008e05b97bfb8f90a6

I've incorporated your changes, however added extra attention to Pet Based classes.

**Comment by Xian55** [comment]
After the recent changes, let me know if the issue regarding pulling enemy faction players still persist.
